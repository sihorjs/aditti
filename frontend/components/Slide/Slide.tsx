import React from "react";
import classnames from "classnames";
import { FeaturedProduct } from "types/carousel";
import "./Slide.less";

type Props = {
    className?: string;
    slide: FeaturedProduct;
};

function Slide(props: Props) {
    return (
        <div className={classnames("slide", props.className)}>
            <div className="slide__img-container">
                <img
                    alt={props.slide.heading}
                    className="slide__img"
                    src={props.slide.imgSrc}
                />
            </div>
            <div className="slide__info">
                <div className="slide__title">{props.slide.heading}</div>
                <div className="slide__description">
                    {props.slide.description}
                </div>
                <button className="slide__btn" name="shop-now" type="button">
                    shop now
                </button>
            </div>
        </div>
    );
}

export default Slide;
