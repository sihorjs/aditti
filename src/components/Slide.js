import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

function Slide(props) {
    return (
        <div className={classnames("slide", props.className)}>
            <div className="slide-img">
                <img src={props.image} alt={props.heading} />
            </div>
            <div className="slide-info">
                <h1>{props.heading}</h1>
                <p>{props.description}</p>
                <button name="shop-now">shop now</button>
            </div>
        </div>
    );
}

Slide.propTypes = {
    className: PropTypes.string,
    description: PropTypes.string.isRequired,
    heading: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
};

Slide.defaultProps = {
    className: null
};

export default Slide;
