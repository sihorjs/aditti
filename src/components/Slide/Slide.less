@import url("less/index.less");

.slide {
    display: flex;

    &__img-container {
        display: flex;
        justify-content: center;
        align-items: center;
        width: 40%;
    }

    &__info {
        width: 60%;
        padding: 40px 0;
        color: @white;
    }

    &__title {
        margin-bottom: 25px;
        font: 60px @heading-font;
    }

    &__description {
        margin-bottom: 20px;
        font: 18px/28px @main-font;
    }

    &__btn {
        width: 130px;
        height: 45px;
        padding: 0;
        border: solid 3px @white;
        font: 27px @heading-font;
        color: @white;
        text-transform: uppercase;
        background-color: transparent;
        border-radius: 10px;
        cursor: pointer;
        outline: none;
    }
}
