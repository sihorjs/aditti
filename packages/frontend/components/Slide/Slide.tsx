import React from "react";
import classnames from "classnames";
import FeaturedProductType from "@aditti/types/carousel";
import "./Slide.less";

type Props = {
    className?: string;
    slide: FeaturedProductType;
};

function Slide(props: Props) {
    return (
        <div className={classnames("slide", props.className)}>
            <div className="slide__img-container">
                <img
                    alt={props.slide.heading}
                    className="slide__img"
                    src={props.slide.imgSrc}
                />
            </div>
            <div className="slide__info">
                <div className="slide__title">{props.slide.heading}</div>
                <div className="slide__description">
                    {props.slide.description}
                </div>
                <button className="slide__btn" name="shop-now" type="button">
                    shop now
                </button>
            </div>
        </div>
    );
}

export default Slide;
