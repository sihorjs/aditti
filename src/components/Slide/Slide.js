import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import "./Slide.less";

function Slide(props) {
    return (
        <div className={classnames("slide", props.className)}>
            <div className="slide__img-container">
                <img
                    alt={props.slide.heading}
                    className="slide__img"
                    src={props.slide.imgSrc}
                />
            </div>
            <div className="slide__info">
                <div className="slide__title">{props.slide.heading}</div>
                <div className="slide__description">
                    {props.slide.description}
                </div>
                <button className="slide__btn" name="shop-now" type="button">
                    shop now
                </button>
            </div>
        </div>
    );
}

Slide.propTypes = {
    className: PropTypes.string,
    slide: PropTypes.shape({
        description: PropTypes.string.isRequired,
        heading: PropTypes.string.isRequired,
        imgSrc: PropTypes.string.isRequired
    }).isRequired
};

Slide.defaultProps = {
    className: null
};

export default Slide;
