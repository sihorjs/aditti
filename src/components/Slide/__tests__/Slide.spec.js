import React from "react";
import { shallow } from "enzyme";
import Slide from "../Slide";

describe("<Slide />", () => {
    const props = {
        heading: "2",
        description:
            "Lorem ipsum dolor sit amet consectetur, adipisicing elit.",
        imgSrc: "img/carousel/t-shirt.png"
    };

    it("renders without crash", () => {
        const tree = shallow(<Slide slide={props} />);

        expect(tree).toMatchSnapshot();
    });

    it("receives className for wrapper from props", () => {
        const className = "slide-wrapper";
        const component = shallow(
            <Slide className={className} slide={props} />
        );

        expect(component.find(".slide").hasClass(className)).toBeTruthy();
    });
});
