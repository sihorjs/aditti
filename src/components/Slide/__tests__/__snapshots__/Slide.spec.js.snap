// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slide /> renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Slide
    className={null}
    slide={
      Object {
        "description": "Lorem ipsum dolor sit amet consectetur, adipisicing elit.",
        "heading": "2",
        "imgSrc": "img/carousel/t-shirt.png",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="slide__img-container"
        >
          <img
            alt="2"
            className="slide__img"
            src="img/carousel/t-shirt.png"
          />
        </div>,
        <div
          className="slide__info"
        >
          <div
            className="slide__title"
          >
            2
          </div>
          <div
            className="slide__description"
          >
            Lorem ipsum dolor sit amet consectetur, adipisicing elit.
          </div>
          <button
            className="slide__btn"
            name="shop-now"
            type="button"
          >
            shop now
          </button>
        </div>,
      ],
      "className": "slide",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="2"
            className="slide__img"
            src="img/carousel/t-shirt.png"
          />,
          "className": "slide__img-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "2",
            "className": "slide__img",
            "src": "img/carousel/t-shirt.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="slide__title"
            >
              2
            </div>,
            <div
              className="slide__description"
            >
              Lorem ipsum dolor sit amet consectetur, adipisicing elit.
            </div>,
            <button
              className="slide__btn"
              name="shop-now"
              type="button"
            >
              shop now
            </button>,
          ],
          "className": "slide__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2",
              "className": "slide__title",
            },
            "ref": null,
            "rendered": "2",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lorem ipsum dolor sit amet consectetur, adipisicing elit.",
              "className": "slide__description",
            },
            "ref": null,
            "rendered": "Lorem ipsum dolor sit amet consectetur, adipisicing elit.",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "shop now",
              "className": "slide__btn",
              "name": "shop-now",
              "type": "button",
            },
            "ref": null,
            "rendered": "shop now",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="slide__img-container"
          >
            <img
              alt="2"
              className="slide__img"
              src="img/carousel/t-shirt.png"
            />
          </div>,
          <div
            className="slide__info"
          >
            <div
              className="slide__title"
            >
              2
            </div>
            <div
              className="slide__description"
            >
              Lorem ipsum dolor sit amet consectetur, adipisicing elit.
            </div>
            <button
              className="slide__btn"
              name="shop-now"
              type="button"
            >
              shop now
            </button>
          </div>,
        ],
        "className": "slide",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="2"
              className="slide__img"
              src="img/carousel/t-shirt.png"
            />,
            "className": "slide__img-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "2",
              "className": "slide__img",
              "src": "img/carousel/t-shirt.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="slide__title"
              >
                2
              </div>,
              <div
                className="slide__description"
              >
                Lorem ipsum dolor sit amet consectetur, adipisicing elit.
              </div>,
              <button
                className="slide__btn"
                name="shop-now"
                type="button"
              >
                shop now
              </button>,
            ],
            "className": "slide__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2",
                "className": "slide__title",
              },
              "ref": null,
              "rendered": "2",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lorem ipsum dolor sit amet consectetur, adipisicing elit.",
                "className": "slide__description",
              },
              "ref": null,
              "rendered": "Lorem ipsum dolor sit amet consectetur, adipisicing elit.",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "shop now",
                "className": "slide__btn",
                "name": "shop-now",
                "type": "button",
              },
              "ref": null,
              "rendered": "shop now",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
